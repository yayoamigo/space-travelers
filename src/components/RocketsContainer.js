import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { fetchRockets } from '../redux/rockets/rockets';
import Rocket from './Rocket';

const RocketsContainer = () => {
  const rockets = useSelector((state) => state.rockets);
  const { length } = rockets;
  const dispatch = useDispatch();
  useEffect(() => {
    if (length === 0) {
      dispatch(fetchRockets());
    }
  }, [dispatch]);
  return (
    <div>
      {
      rockets.map((rocket) => (
        <Rocket
          key={rocket.id}
          name={rocket.rocket_name}
          description={rocket.description}
          image={rocket.flickr_images}
          id={rocket.id}
          reserved={rocket.reserved}
        />
      ))
    }
    </div>
  );
};

export default RocketsContainer;
