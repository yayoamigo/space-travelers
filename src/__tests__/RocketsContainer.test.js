// import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import RocketsContainer from '../components/RocketsContainer';

describe('RocketsContainer component test', () => {
  test('should render', () => {
    const rocketsContainer = renderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <RocketsContainer />
        </BrowserRouter>
        ,
      </Provider>,
    );
    const tree = rocketsContainer.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
