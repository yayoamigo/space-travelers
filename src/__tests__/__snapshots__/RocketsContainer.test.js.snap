// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RocketsContainer component test should render 1`] = `
Array [
  <div />,
  ",",
]
`;
